include/ck_tile/host/host_tensor.hpp

```diff
@@ -155,19 +160,34 @@ struct HostTensorDescriptor
         return space;
     }
 
-    const std::vector<std::size_t>& get_lengths() const { return mLens; }
-    const std::vector<std::size_t>& GetStrides() const { return mStrides; }
+    std::size_t get_length(std::size_t dim) const { return mLens[dim]; }
+
+    auto get_lengths() const
+    {
+        using iterator = remove_cvref_t<decltype(mLens)>::const_iterator;
+        return iterator_range<iterator>(mLens);
+    }
+
+    std::size_t get_stride(std::size_t dim) const { return mStrides[dim]; }
+
+    auto get_strides() const
+    {
+        using iterator = remove_cvref_t<decltype(mStrides)>::const_iterator;
+        return iterator_range<iterator>(mStrides);
+    }
 
     template <typename... Is>
-    std::size_t GetOffsetFromMultiIndex(Is... is) const
+    std::enable_if_t<((std::is_integral_v<Is> && std::is_convertible_v<Is, std::size_t>)&&...),
+                     std::size_t>
+    GetOffsetFromMultiIndex(Is... is) const
```

Comment:
Do we need to sync the naming style of funcion?

Reply:
sure, I will rename the functions